libs/community/langchain_google_community/docai.py

```diff
@@ -153,14 +166,37 @@ def online_process(
                 "documentai_toolbox package not found, please install it with "
                 "`pip install langchain-google-community[docai]`"
             ) from exc
-        ocr_config = (
-            OcrConfig(enable_native_pdf_parsing=enable_native_pdf_parsing)
-            if enable_native_pdf_parsing
-            else None
-        )
-        individual_page_selector = (
-            IndividualPageSelector(pages=page_range) if page_range else None
-        )
+
+        if self._use_layout_parser:
+            layout_config = ProcessOptions.LayoutConfig(
+                chunking_config=ProcessOptions.LayoutConfig.ChunkingConfig(
+                    chunk_size=chunk_size,
+                    include_ancestor_headings=include_ancestor_headings,
+                )
+            )
+            individual_page_selector = (
+                ProcessOptions.IndividualPageSelector(pages=page_range)
+                if page_range
+                else None
+            )
+            process_options = ProcessOptions(
+                layout_config=layout_config,
+                individual_page_selector=individual_page_selector,
+            )
```

Comment:
Maybe some of this could be brought out into a function to re-use for the other methods like batch processing.